x-pack/plugins/apm/server/routes/services/get_services_detailed_statistics/index.ts

```diff
@@ -26,30 +27,29 @@ export async function getServicesDetailedStatistics({
   kuery: string;
   setup: Setup;
   searchAggregatedTransactions: boolean;
+  searchAggregatedServiceMetrics: boolean;
   offset?: string;
   start: number;
   end: number;
   randomSampler: RandomSampler;
 }) {
-  return withApmSpan('get_service_detailed_statistics', async () => {
-    const commonProps = {
-      serviceNames,
-      environment,
-      kuery,
-      setup,
-      searchAggregatedTransactions,
-      start,
-      end,
-      randomSampler,
-    };
-
-    const [currentPeriod, previousPeriod] = await Promise.all([
-      getServiceTransactionDetailedStatistics(commonProps),
-      offset
-        ? getServiceTransactionDetailedStatistics({ ...commonProps, offset })
-        : Promise.resolve({}),
-    ]);
-
-    return { currentPeriod, previousPeriod };
-  });
+  const commonProps = {
+    serviceNames,
+    environment,
+    kuery,
+    setup,
+    start,
+    end,
+    randomSampler,
+    offset,
+  };
+  return searchAggregatedServiceMetrics
+    ? getServiceAggregatedDetailedStatsPeriods({
```

Comment:
IMHO it's unfortunate that we now have separate functions for these, if event.outcome was part of the grouping key we could have one function (that aggregates over transaction.duration.us, transaction.duration.histogram, or transaction.duration.summary).

Reply:
I agree, we can talk with @Mpdreamz and apm-server team to see if we can update it